code/script.js

```diff
@@ -30,18 +53,130 @@ const showMessage = (message, sender) => {
   chat.scrollTop = chat.scrollHeight
 }
 
-// Starts here
+// welcome message, ask for user's name
 const greeting = () => {
-  showMessage(`Hello there, What's your name?`, 'bot')
-  // Just to check it out, change 'bot' to 'user' here 👆
+  showMessage(`Hello there, what is your name?`, 'bot')
 }
 
-// Set up your eventlisteners here
-
-// When website loaded, chatbot asks first question.
-// normally we would invoke a function like this:
-// greeting()
-// But if we want to add a little delay to it, we can wrap it in a setTimeout:
-// setTimeout(functionName, timeToWaitInMilliSeconds)
-// This means the greeting function will be called one second after the website is loaded.
 setTimeout(greeting, 1000)
+
+// Bot validate user name
+let nameUser = null;
+
+nameForm.addEventListener("submit", (event) => {
+  event.preventDefault()
+  nameUser = document.querySelector("#name-input").value
+  if (nameUser.length === 0 || nameUser === null) {
+    showMessage('Please tell me your name', 'bot')
+  }
+  else {
+    showMessage(nameUser, "user")
+    setTimeout(showGenre, 500)
+  }
+})
+
+// show genre (show buttons) bot ask user to pick a genre
```

Comment:
Your buttons code should be stored here and render dynamically only when needed. It is a very easy process to uncover hidden elements. 🤓